src/hotspot/share/runtime/safepointMechanism.cpp

```diff
@@ -141,7 +160,6 @@ void SafepointMechanism::process(JavaThread *thread, bool allow_suspend, bool ch
   } while (need_rechecking);
 
   update_poll_values(thread);
-  OrderAccess::cross_modify_fence();
```

Comment:
Has this simply been moved to cover more paths?

Reply:
The new logic inside of update_poll_values makes is more accurate and made this cross_modify_fence redundant. That's why I removed it.

Reply:
> Has this simply been moved to cover more paths?

I removed it because it is redundant after updating update_poll_values to more precisely identify when cross_modify_fence should be called. It will then already have been called inside of update_poll_values, when it was needed here at the call site.

Reply:
Okay.